scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
I agree that is inconsistent.  As far as I can see, the MATLAB code uses `g` format for both.  If we change it, it will be backwards incompatible:

```
In [7]: '%.5g' % 1.00001
Out[7]: '1'

In [8]: '%.5e' % 1.00001
Out[8]: '1.00001e+00'
```

@rgommers  - do you have any opinion on this?  Maybe it's reasonable to match MATLAB by using `g` throughout, and add string formatting to allow decimal places etc if the user selects them.